smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy

```diff
@@ -243,4 +219,59 @@ abstract class SmokeTest extends Specification {
     }
     return encoding
   }
+
+  static class Backend {
+    private static final INSTANCE = new Backend()
+
+    private final Network network = Network.newNetwork()
+    private GenericContainer backend
+    private GenericContainer collector
+
+    boolean started = false
+
+    static Backend getInstance() {
+      return INSTANCE
+    }
+
+    def setup() {
+      // we start backend & collector once for all tests
+      if (started) {
+        return
+      }
+      started = true
+
+      backend = new GenericContainer<>("ghcr.io/open-telemetry/java-test-containers:smoke-fake-backend-20201128.1734635")
+        .withExposedPorts(8080)
+        .waitingFor(Wait.forHttp("/health").forPort(8080))
+        .withNetwork(network)
+        .withNetworkAliases("backend")
+        .withImagePullPolicy(PullPolicy.alwaysPull())
+        .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("smoke.tests.backend")))
+      backend.start()
+
+      collector = new GenericContainer<>("otel/opentelemetry-collector-dev:latest")
+        .dependsOn(backend)
+        .withNetwork(network)
+        .withNetworkAliases("collector")
+        .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("smoke.tests.collector")))
+        .withImagePullPolicy(PullPolicy.alwaysPull())
+        .withCopyFileToContainer(MountableFile.forClasspathResource("/otel.yaml"), "/etc/otel.yaml")
+        .withCommand("--config /etc/otel.yaml")
+      collector.start()
+    }
+
+    int getMappedPort(int originalPort) {
+      return backend.getMappedPort(originalPort)
+    }
+
+    def cleanup() {
+    }
+
+    // currently unused
+    def stop() {
```

Comment:
Nit: consider adding a shutdown hook for stoping those docker containers

Reply:
shutdown hook added